evm-validate/src/lib.rs

```diff
@@ -417,3 +468,764 @@ fn validate_transaction_hashes(
+    let mut block_tx_receipts = Vec::<ReceiptEnvelope>::with_capacity(200);
+    // initialize an empty vec of logs, used if the tx failed or doesn't have logs
+    let empty_logs = Vec::<Log>::new();
+
+    let tx_iterators = tx_array
+        .block_number.iter()
+        .zip(tx_array.gas_limit.iter())
+        .zip(tx_array.gas_price.iter())
+        .zip(tx_array.hash.iter())
+        .zip(tx_array.input.iter())
+        .zip(tx_array.nonce.iter())
+        .zip(tx_array.to.iter())
+        .zip(tx_array.tx_index.iter())
+        .zip(tx_array.value.iter())
+        .zip(tx_array.v.iter())
+        .zip(tx_array.r.iter())
+        .zip(tx_array.s.iter())
+        .zip(tx_array.max_priority_fee_per_gas.iter())
+        .zip(tx_array.max_fee_per_gas.iter())
+        .zip(tx_array.chain_id.iter())
+        .zip(tx_array.cumulative_gas_used.iter())
+        .zip(tx_array.contract_address.iter())
+        .zip(tx_array.logs_bloom.iter())
+        .zip(tx_array.tx_type.iter())
+        .zip(tx_array.status.iter())
+        .zip(tx_array.sighash.iter())
+        .zip(tx_array.access_list.iter())
+        .zip(tx_array.max_fee_per_blob_gas.iter())
+        .zip(tx_array.blob_versioned_hashes.iter());
+    
+    // iterate over transactions rows
+    for (((((((((((((((((((((((
+        tx_block_nums_opt
+        , tx_gas_limit_opt)
+        , tx_gas_price_opt)
+        , tx_hash_opt)
+        , tx_input_opt)
+        , tx_nonce_opt)
+        , tx_to_opt)
+        , tx_tx_idx_opt)
+        , tx_value_opt)
+        , tx_v_opt)
+        , tx_r_opt)
+        , tx_s_opt)
+        , tx_max_priority_fee_per_gas_opt)
+        , tx_max_fee_per_gas_opt)
+        , tx_chain_id_opt)
+        , tx_cumulative_gas_used_opt)
+        , tx_contract_address_opt)
+        , tx_logs_bloom_opt)
+        , tx_type_opt) 
+        , tx_status_opt)
+        , tx_sighash_opt)
+        , tx_access_list_opt)
+        , tx_max_fee_per_blob_gas_opt)
+        , tx_blob_versioned_hashes_opt) in tx_iterators {
+
+        // cast values to expected types
+        let block_num = tx_block_nums_opt.unwrap();
+        let gas_limit = tx_gas_limit_opt.unwrap();
+        let gas_price = U256::try_from_be_slice(tx_gas_price_opt.unwrap()).expect("invalid gas price");
+        let gas_price = gas_price.try_into().unwrap();
+        let expected_hash: FixedBytes<32> = tx_hash_opt.unwrap().try_into().unwrap();
+        let input = tx_input_opt.unwrap();
+        let input = Bytes::copy_from_slice(input);
+        let nonce = tx_nonce_opt.unwrap();
+        let to:Option<Address> = tx_to_opt.map(|a| a.try_into().unwrap());
+        let tx_idx = tx_tx_idx_opt.unwrap();
+        let value = U256::try_from_be_slice(tx_value_opt.unwrap()).expect("invalid value");
+        let v = match tx_v_opt.unwrap(){
+            [0] | [27] | [37] => false,
+            [1] | [28] | [38] => true,
+            _ => return Err(anyhow!("invalid v")),
+        };
```

Comment:
This is working by try-and-error. I don't understand why `v` can have these other values.

Reply:
check this out https://ethereum.stackexchange.com/questions/13778/get-public-key-of-any-ethereum-account

maybe 37 and 38 are not needed?